### PR TITLE
Internal-tools-authentication

### PR DESCRIPTION
**Pull request by jhylisa666**
Added authentication for internal tools. 
Added sign-in and sign-out using Google Authentication. 
Acceptance criteria:

 User cannot access any page of our tools application unless they are logged in
 User can sign in via email if not logged in and sign out if logged in
 Only users with the an email within our domain (@hackbeanpot.com) can login. Otherwise, display an error indicating the user is not authorized to view our tools.

https://user-images.githubusercontent.com/92692037/174447040-288834fb-eebc-4fd3-882f-d9fc381c71c7.mov

** Styling is off but is fixed in Judy and Dhivas's styling PR ^^

### CODE REVIEW

File: pages/_app.tsx [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
 import type { AppProps } from 'next/app'
+import { SessionProvider } from 'next-auth/react'
+
+import React from 'react'
 
 function MyApp ({ Component, pageProps }: AppProps) {
-  return <Component {...pageProps} />
+  return (
+    <SessionProvider
+      // Provider options are not required but can be useful in situations where
```
Comment (jzhzhang66) [inline pages/_app.tsx:12]: Remove all comments

File: pages/api/auth/[...nextauth].js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import NextAuth from 'next-auth'
+import GoogleProvider from 'next-auth/providers/google'
+
+// dynamic route handler
```
Comment (jzhzhang66) [inline pages/api/auth/[...nextauth].js:4]: Remove all comments in files

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]
It's working for me! The styling is fixed in the other PR so this PR looks good to me! 

Just remove the comments and then I'm ready to approve. Thanks!

**Review by Dhivas-Sugumar — COMMENTED** [review]
Works great for me! Approved

File: pages/_app.tsx (hunk 1 above)
Comment (jhylisa666) [reply pages/_app.tsx:12]: just removed!

**Review by jzhzhang66 — APPROVED** [review]
QUEEENS

**Pull request merged**